### PR TITLE
challenge_26 solutions

### PR DESCRIPTION
**Pull request by Lastrophysicien**
---
name: Prime Number Challenge
about: A template PR for code review with a checklist
---

## Behavior
"""
    The prime_number function determines if a number is a prime number. It takes an integer as input and returns a string indicating whether the number is "is prime" or "not prime". The function first checks if the number is less than or equal to 1, immediately returning "not prime" if true. It then iterates from 2 to the square root of the number, checking for divisors. If a divisor is found, it returns "not prime". If no divisors are found, it returns "is prime". The script includes an example usage where the user is prompted to enter a number, and the function's result is printed.
"""

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/challenge_26/prime_numbers.py
Comment (AnaiMurillo) [inline solutions/challenge_26/prime_numbers.py:1]: The function file includes an `if __name__ == "__main__"`: block that calls the function. Please remove this execution. The function file should focus only on the definition.

File: solutions/challenge_26/prime_numbers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if number % i == 0:
+            return "not prime"
+    return "is prime"
+
+
+# Example usage
+if __name__ == "__main__":
+    number = int(input("Enter a number: "))
+    result = prime_numbers(number)
+    print(result)
```
Comment (AnaiMurillo) [inline solutions/challenge_26/prime_numbers.py]: The code should not have ` print ` statements anywhere

**Review by AnaiMurillo — CHANGES_REQUESTED** [review]
You have done a great job!! Only apply those little changes and we are done :)

**Review by AnaiMurillo — APPROVED** [review]
Everything is fine, good job! :)
